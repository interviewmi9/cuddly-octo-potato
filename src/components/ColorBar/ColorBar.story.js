import React from 'react'
import styled from 'styled-components'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import ColorBar from './'

const Wrapper = styled.div`
  height: 50px;
  width: 7px;
`

const colorIds = [3, 2, 1]
storiesOf('ColorBar', module)
  .addDecorator((story, context) => withInfo()(story)(context))
  .add('Default view', () => {
    return <Wrapper><ColorBar colorIds={colorIds} /></Wrapper>
  })
