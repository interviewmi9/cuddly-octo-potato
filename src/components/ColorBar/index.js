import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { colors } from '../../fixtures'

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
`

const Bar = styled.div`
  background-color: ${({ backgroundColor }) => backgroundColor};
  flex-grow: 1;
`

const ColorBar = ({ colorIds }) => {
  return (
    <Wrapper>
      {colorIds.map(id => <Bar key={id} backgroundColor={colors[id].value} />)}
    </Wrapper>
  )
}

ColorBar.propTypes = {
  colorIds: PropTypes.arrayOf(PropTypes.number).isRequired,
}

export default ColorBar
