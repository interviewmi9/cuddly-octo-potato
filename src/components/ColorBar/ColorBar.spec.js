import React from 'react'
import ColorBar from './'

describe('Color Bar component', () => {
  const colorIds = [1, 2, 3]

  it('renders correctly', () => {
    const rendered = render(<ColorBar colorIds={colorIds} />)
    expect(rendered).toMatchSnapshot()
  })
})
