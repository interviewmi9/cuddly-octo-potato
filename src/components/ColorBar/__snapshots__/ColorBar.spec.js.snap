// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Bar component renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

.c1 {
  background-color: #b2d9f1;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  background-color: #d7efa5;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  background-color: #f9e9c0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
</div>
`;
